src/main/java/io/jenkins/plugins/sample/OnboardingTaskBuilder.java

```diff
@@ -0,0 +1,108 @@
+package io.jenkins.plugins.sample;
+
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.AbstractProject;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.Builder;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import jenkins.model.GlobalConfiguration;
+import jenkins.tasks.SimpleBuildStep;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class OnboardingTaskBuilder extends Builder implements SimpleBuildStep {
+
+    private final String name;
+    private String selectedCategory;
+    @DataBoundConstructor
+    public OnboardingTaskBuilder(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    @DataBoundSetter
+    public void setSelectedCategory(String category) {
+         this.selectedCategory = category;
+    }
```

Comment:
Missing a getter.

Reply:
Adding getter resolved this. I was able to get generated script like below
```
onboarding name: 'sa', selectedCategory: 'b5d06f12-6446-47de-8abb-143380ed30af'

So far I didn't need to expose gettter as I was selecting the categories from dropdown which is populated from global config's data and selectedCategory is used within the class OnboardingTaskBuilder. I had not added getter as it was not required. 
Is it always recommended to add getters even if a property is not being used (atleast currently) outside the object's context?

Reply:
If you expect the snippet generator to work then you must have getters for all fields. (Or the fields themselves can be `public.`)

Reply:
Okay, thanks.
 PS: In my opinion, snippet generator could generate the snippet considering only the properties which are public (either via getter or public) and ignoring the private properties. That would be less confusing (atleast to newcomers).

Reply:
Discussed with Thor and came to a conclusion tha it was failing for a case where you have data bound setter but not the getter. Private properties without public scope (getter and setter) are ignored as expected.